autorest/client.go

```diff
@@ -262,3 +311,80 @@ func (c Client) ByInspecting() RespondDecorator {
 	}
 	return c.ResponseInspector
 }
+
+func withLogging() PrepareDecorator {
+	return func(p Preparer) Preparer {
+		return PreparerFunc(func(r *http.Request) (*http.Request, error) {
+			// execute any remaining preparers before logging
+			r, err := p.Prepare(r)
+			if logLevel > logLevelNone {
+				b := &bytes.Buffer{}
+				fmt.Fprintf(b, "(%s) REQUEST: %s %s\n", time.Now().Format(logTimestamp), r.Method, r.URL.String())
+				// dump headers
+				for k, v := range r.Header {
+					if strings.ToLower(k) == "authorization" {
+						v = []string{"**REDACTED**"}
+					}
+					fmt.Fprintf(b, "%s: %s\n", k, strings.Join(v, ","))
+				}
+				ct := r.Header.Get(headerContentType)
+				if logLevel == logLevelFull && r.Body != nil && strings.Index(ct, mimeTypeOctetStream) == -1 {
+					// dump body
+					var body []byte
+					body, err = ioutil.ReadAll(r.Body)
+					if err == nil {
+						fmt.Fprintln(b, string(body))
+						if nc, ok := r.Body.(NopClosingReadSeeker); ok {
```

Comment:
Since you only need `Seek`, I'd assert just on [io.Seeker](https://godoc.org/io#Seeker).

Reply:
Derp, and I was wracking my brain as I really didn't want to add ```NopClosingReadSeeker```.  I guess the obvious answers are really hard to find sometimes.  :p

Reply:
I've been there before, it's incredible what other people's eyes see. 😝